airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -377,29 +377,32 @@ def _check_dataflow_job_state(self, job) -> bool:
         :rtype: bool
         :raise: Exception
         """
-        if self._wait_until_finished is None:
-            wait_for_running = job['type'] == DataflowJobType.JOB_TYPE_STREAMING
```

Comment:
what do you think about just using `wait_for_running = job.get('type', '') == DataflowJobType.JOB_TYPE_STREAMING` here?

I am worried that skipping rest of the logic to wait for the job type may hang it forever.
I can imagine situation where flex template job may be cancelled before state appear in payload.

Another case is when job type is not necessary to return True:
```
            elif job['currentState'] in DataflowJobStatus.AWAITING_STATES:
                return self._wait_until_finished is False
```

cc @mik-laj

Reply:
agree - this is a better way, have changed for review.